Client/Frontend/Widgets/PhotonActionSheetProtocol.swift

```diff
@@ -159,6 +161,28 @@ extension PhotonActionSheetProtocol {
             }
         }
         
+        let pinToTopSites = PhotonActionSheetItem(title: Strings.PinTopsiteActionTitle, iconString: "action_pin") { action in
+            guard let url = tab.url?.displayURL else { return }
+            let absoluteString = url.absoluteString
+            
+            let sql = self.profile.favicons as! SQLiteHistory
```

Comment:
I know `profile.favicons` and `profile.history` is the same thing but reading this code it feels weird that this is `profile.favicons`. History makes a bit more sense here. "Oh this is fetching a site from history"

Reply:
This should never fail, but lets not force unwrap. You could add this to the guard above.